scm/driver/github/milestone.go

```diff
@@ -0,0 +1,111 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/drone/go-scm/scm"
+)
+
+type milestoneService struct {
+	client *wrapper
+}
+
+type milestone struct {
+	ID           int       `json:"id"`
+	Number       int       `json:"number"`
+	Title        string    `json:"title"`
+	Description  string    `json:"description"`
+	State        string    `json:"state"`
+	DueOn        time.Time `json:"due_on"`
+	URL          string    `json:"url"`
+	HTMLURL      string    `json:"html_url"`
+	LabelsURL    string    `json:"labels_url"`
+	Creator      user      `json:"creator"`
+	OpenIssues   int       `json:"open_issues"`
+	ClosedIssues int       `json:"closed_issues"`
+	NodeID       string    `json:"node_id"`
+	CreatedAt    time.Time `json:"created_at"`
+	UpdatedAt    time.Time `json:"updated_at"`
+	ClosedAt     time.Time `json:"closed_at"`
+}
+
+type milestoneInput struct {
+	Title       string    `json:"title"`
+	State       string    `json:"state"`
+	Description string    `json:"description"`
+	DueOn       time.Time `json:"due_on"`
```

Comment:
I think `due_on` needs to support a nul value, but for this we should use `null.Time`. See https://github.com/drone/go-scm/pull/121#discussion_r715715079